packages/story-editor/src/app/media/utils/useFFmpeg.js

```diff
@@ -282,6 +282,65 @@ function useFFmpeg() {
     [getFFmpegInstance]
   );
 
+  /**
+   * Segment a video using FFmpeg.
+   *
+   * @param {File} file Original video file object.
+   * @param {string} segmentTime number of secs to split the video into.
+   * @return {Promise<File[]>} Segmented video files .
+   */
+  const segmentVideo = useCallback(
+    async (file, segmentTime) => {
+      let ffmpeg;
+      try {
+        ffmpeg = await getFFmpegInstance(file);
+        const type = file?.type || MEDIA_TRANSCODED_MIME_TYPE;
+        const ext = getExtensionFromMimeType(type);
+        const outputFileName = getFileBasename(file) + '_%03d.' + ext;
+        await ffmpeg.run(
+          '-i',
+          file.name,
+          '-c',
+          'copy',
+          '-f',
+          'segment',
+          '-segment_time',
+          `${segmentTime}`,
+          '-reset_timestamps',
+          '1',
+          outputFileName
+        );
+
+        const files = [];
+
+        await ffmpeg
+          .FS('readdir', '/')
+          .filter(
+            (outputFile) =>
+              outputFile !== file.name && outputFile.endsWith(`.${ext}`)
+          )
+          .forEach(async (outputFile) => {
+            const data = await ffmpeg.FS('readFile', outputFile);
+            files.push(
+              blobToFile(new Blob([data.buffer], { type }), outputFile, type)
+            );
+          });
+        return files;
+      } catch (err) {
+        // eslint-disable-next-line no-console -- We want to surface this error.
+        console.error(err);
```

Comment:
Not console, track error.

Reply:
> Screen.Recording.2022-09-22.at.11.25.31.mov 
> In my testing, I found two things.
> 
> * It seems like some of the segments are 19 / 21 seconds. This seems odd, worth looking into.
> * When the page is created, it automatically changes to the new page and starts playing the video. This makes a lot of noise.


> It seems like some of the segments are 19 / 21 seconds. This seems odd, worth looking into.

I added a note about that under testing in the summary above

 👉 Note: the video is segmented based on the videos keyframes -- we can force keyframes to happen at more exact intervals but this will result in larger videos --- as the keyframes we would be forcing to ensure the segment length are for our purposes vs the actual file compression (more keyframes = larger file).  In this case the keyframe are frame `diff` information not "animation" keyframes.
 
> When the page is created, it automatically changes to the new page and starts playing the video. This makes a lot of noise.

Thinking the same --- I'm using the existing logic to add pages .... do we have a way to insert a new page without moving to it?

Reply:
This is in line with the rest of the functions in this file.